[7540] Add component to display AI report text with dummy data for now

philli-m opened the pull request.

philli-m commented on apps/userdashboard/assets/js/a4_candy_userdashboard/ModerationNotification.jsx:
```diff
@@ -302,6 +305,7 @@ export const ModerationNotification = (props) => {
           </div>}
 
         <p>{commentText}</p>
+        <AIFeedback AIReport={translated.dummyText} />
```
this will actually come from the serializer but until we have it I just added a string

m4ra replied:
A semantic note: Hannes started by namimg the module AiReport. It is helpful to remain consistent because when we are looking up AiReport we find the instances of this name. So either we all change it to AIReport/AIFeedback, or you change it here to AiReport/AiFeedback @goapunk @hklarner

m4ra reviewed (commented):
LGTM.  
It would be great if we have a docs/ai_report.md with the structure of the files that contain relevant code. Me and Hannes can update it with the backend files we are changing. See my comment here: https://github.com/liqd/adhocracy-plus/pull/2510#discussion_r1281495526

philli-m commented:
> LGTM. It would be great if we have a docs/ai_report.md with the structure of the files that contain relevant code. Me and Hannes can update it with the backend files we are changing. See my comment here: [liqd/adhocracy-plus#2510 (comment)](https://github.com/liqd/adhocracy-plus/pull/2510#discussion_r1281495526)

@m4ra I just added the file with a FIXME as a reminder but I wasn't quite sure what to add yet as it's purely a dummy component as it stands. I also added a changelog for the commit and updated the name

m4ra commented:
@philli-m 

> @m4ra I just added the file with a FIXME as a reminder but I wasn't quite sure what to add yet as it's purely a dummy component as it stands. I also added a changelog for the commit and updated the name

my suggestion is to at least add the paths to the files we are changing for now as notes for later. E.x 
```
react components
apps/userdashboard/assets/js/a4_candy_userdashboard/ 
```

goapunk reviewed (changes requested):
Looks good to me, just one little thing: the padding to the right for the text is too small compared to the design (to the left it's correct 40px) but I assume because of the exclamation mark icon the text is moved more to the right so the padding becomes too small

goapunk commented on apps/userdashboard/assets/js/a4_candy_userdashboard/AiReportExplanation.jsx:
```diff
@@ -0,0 +1,41 @@
+        className="fas fa-exclamation-circle text-danger pt-1 pe-2"
+        aria-hidden="True"
+      />
+      <p>
```
in the design the intro and the report text are text size 14px, not the default one. Did you intentionally leave this out till we get the proper text?

philli-m replied:
no just didn't notice

goapunk commented:
@philli-m  I added a small test (but also realized it probably isn't of that much use until we have the real functionality).

philli-m commented on apps/userdashboard/assets/js/a4_candy_userdashboard/__tests__/AiReportExplanation.jest.jsx:
```diff
@@ -0,0 +1,37 @@
+  const button = screen.getByRole('button')
+  fireEvent.click(button)
+  expect(container).toMatchSnapshot()
+})
```
@goapunk how are these testings the read more or less? they don't seem to check if the text changes, unless I am missing something?
